client/static/app.js

```diff
@@ -0,0 +1,86 @@
+const searchbutton = document.querySelector('#searchbutton');
+const form = document.querySelector('form');
+const searchbar = document.querySelector('#searchbar');
+const lucky = document.getElementById('lucky');
+const resultsArea = document.querySelector('#results');
+// const randLink = document.getElementById('randLink');
+
+async function pageLoad(){
+    resultsArea.textContent = "";
+    const url = "http://localhost:4000/results"
+    await fetch(url)
+    .then(res => res.json())
+    .then(results => {
+        const num = Math.floor(Math.random() * results.length);
+        const randomChoice = results[num];
+        const randomLink = randomChoice.link;
+        console.log(randomLink);
+        const anchorTag = randomLink;
+        lucky.setAttribute("href", anchorTag);
+        console.log(lucky);
+
+})
+}
+
+form.addEventListener("submit", e=>{
+    resultsArea.textContent = "";
+    e.preventDefault();
+    const searchData = new FormData(form);
+    const entries = searchData.entries();
+    const data = Object.fromEntries(entries);
+    console.log(data.searchbar)
+    const url = "http://localhost:4000/results"
+    fetch(url)
```

Comment:
I would probably put the `fetch(url)` in its own function and add `.catch` to check for errors when making the http request.